subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -639,6 +640,23 @@ static int action_set(const struct bt_mesh_model *model, struct bt_mesh_msg_ctx
 		run_scheduler(srv);
 	}
 
+	if ((srv->sch_reg[idx].action == BT_MESH_SCHEDULER_NO_ACTIONS) &&
+	    (srv->active_bitmap & BIT(idx))) {
+		WRITE_BIT(srv->active_bitmap, idx, 0);
+
+		bool reschedule = srv->idx == idx;
+
+		if (srv->active_bitmap == 0) {
+			srv->idx = BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT;
+			srv->last_idx = BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT;
+		}
+
+		/* If the next scheduled action is set to "NO_ACTIONS", schedule the next in line */
+		if (reschedule) {
+			run_scheduler(srv);
```

Comment:
You need to add delayed work cancelation after this: https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/bluetooth/mesh/scheduler_srv.c#L401